hdKeysGenerator/main.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"context"
+	"os"
+	"sort"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-tools-go/hdKeysGenerator/common"
+	"github.com/ElrondNetwork/elrond-tools-go/hdKeysGenerator/export"
+	"github.com/urfave/cli"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	appVersion = "1.0.0"
+)
+
+var log = logger.GetOrCreate("main")
+
+func main() {
+	app := cli.NewApp()
+	cli.AppHelpTemplate = helpTemplate
+	app.Version = appVersion
+	app.Name = "HD keys generator app"
+	app.Usage = "Tool for generating (deriving) HD keys from a given mnemonic"
+	app.Flags = getAllCliFlags()
+	app.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+
+	app.Action = generateKeys
+
+	start := time.Now()
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+
+	log.Info("Done in", "seconds", time.Since(start).Seconds())
+}
+
+func generateKeys(ctx *cli.Context) error {
```

Comment:
I would move the logic of generateKeys & generateKeysInParallel in another part of the program.